backend/config/config.go

```diff
@@ -217,7 +221,12 @@ func NewConfig(appConfigFilename string, accountsConfigFilename string) *Config
 		accountsConfig:         newDefaultAccountsonfig(),
 	}
 	config.load()
-	return config
+	var err error
+	config.appConfig, err = config.checkAPIKeys(config.appConfig)
```

Comment:
Code reuse: use `config.SetAppConfig()` which sets and saves `config.appConfig.`

Naming: `checkAPIKeys` does not check, but modify the config. please use `updateInfuraAPIKeys()` or similar.

Reply:
Well, it checks and then modifies if the check fails :)
But you're right, `update` works better

Reply:
Updated.